### PR TITLE
Execute `EnhancedGraphTable#queueRepositoryUpdateAndModelRefresh` mostly outside of UI thread + ditch `WriteActionUtils#blockingRunWriteActionOnUIThread`

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/BaseSlideInBackgroundable.java [hunk 1]
```diff
@@ -87,7 +87,7 @@ public void run(ProgressIndicator indicator) {
       return;
     }
 
-    runWriteActionOnUIThread(() -> {
```
Comment (PawelLipski) [inline frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/backgroundables/BaseSlideInBackgroundable.java:90]: Motivation: decrease the risk for race conditions

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 2]
```diff
@@ -509,43 +509,43 @@ public void queueRepositoryUpdateAndModelRefresh(@UI Runnable doOnUIThreadWhenRe
       return;
     }
 
-    if (!project.isDisposed()) {
-      ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> {
```
Comment (PawelLipski) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:513]: Motivation: make the flow easier to reason about

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java [hunk 3]
```diff
@@ -509,43 +509,43 @@ public void queueRepositoryUpdateAndModelRefresh(@UI Runnable doOnUIThreadWhenRe
...
-          }
-        };
+    if (project.isDisposed()) {
+      LOG.debug("Project is disposed");
+      return;
+    }
 
-        setTextForEmptyTable(getString("string.GitMachete.EnhancedGraphTable.empty-table-text.loading"));
+    ModalityUiUtil.invokeLaterIfNeeded(NON_MODAL, () -> {
+      setTextForEmptyTable(getString("string.GitMachete.EnhancedGraphTable.empty-table-text.loading"));
```
Comment (PawelLipski) [inline frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java:518]: Just this piece is scheduled onto UI thread

File: frontend/base/src/main/java/com/virtuslab/gitmachete/frontend/common/WriteActionUtils.java [hunk 4]
```diff
@@ -10,10 +10,7 @@ public final class WriteActionUtils {
 
   private WriteActionUtils() {}
 
-  public static <E extends Throwable> void runWriteActionOnUIThread(@UI ThrowableRunnable<E> action) {
-    ApplicationManager.getApplication().invokeLater(getWriteActionRunnable(action));
-  }
-
+  // We don't provide asynchronous (non-blocking) variant since it turned out prone to race conditions.
```
Comment (mkondratek) [inline frontend/base/src/main/java/com/virtuslab/gitmachete/frontend/common/WriteActionUtils.java:13]: 👍

**Review by mkondratek — APPROVED** [review]
fine

**Pull request merged**